pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -451,6 +455,48 @@ public String getSubtaskProgress(@Context HttpHeaders httpHeaders,
     }
   }
 
+  @GET
+  @Path("/tasks/subtask/state/{subTaskState}/progress")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation("Get progress of all subtasks with specified state tracked by minion worker in memory")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public String getSubtaskWithGivenStateProgress(@Context HttpHeaders httpHeaders,
```

Comment:
Do you think that it's better to group the result by taskName? 
```
workerid -> taskName -> subtasks
```

Or, we may want to filter based on the taskName as well?

Reply:
1. Personally, I don't think it's necessary to add taskName key, because 

(1) subtask names have all information include: task type, task name, sub task id 
(2) if we are going to add `taskName` key, are we going to also add `task type` key? It will become too complex

2. As for now, it is not needed to filter based on the taskName. And the current implementation is easy to extend to add taskName filter. We can also add more filters in the future.

What do you think?

Reply:
sounds good to me.